test/resolver/ecto_test.exs

```diff
@@ -0,0 +1,99 @@
+defmodule Resolver.EctoTest do
+  use ExUnit.Case
+  alias Backoffice.Resolvers.Ecto, as: EctoResolver
+  require Ecto.Query
+
+  defmodule Todo do
+    use Ecto.Schema
+
+    @primary_key false
+    schema "todos" do
+      field(:userId, :string)
+      field(:id, :string)
+      field(:completed, :boolean)
+      field(:title, :string)
+    end
+  end
+
+  defmodule MockRepo do
+    def all(query) do
+      send(self(), %{
+        order_bys: query.order_bys |> Enum.map(& &1.expr),
+        preload: query.preloads
+      })
+
+      [%Todo{}]
+    end
+
+    # This is called by load to find out how many entries there are
```

Comment:
Why we need the `one` function: 
The pagination function calls an aggregate count on the number of items that exist for a resource ([here](https://github.com/edisonywh/backoffice/blob/d310831a4a3018d3f9ee5dea17aa7849fcc9e71a/lib/backoffice/resolvers/ecto.ex#L217)). That aggregate count is piped into a `repo.one`.

This means that our mock repo must also provide a `one/1` function.

---

Why does it return 1: 
We can't use 0 or nil because this wuld result in 0 `total_pages` in the [paginate function](https://github.com/edisonywh/backoffice/blob/d310831a4a3018d3f9ee5dea17aa7849fcc9e71a/lib/backoffice/resolvers/ecto.ex#L192) which would short-circuit the [entries function](https://github.com/edisonywh/backoffice/blob/d310831a4a3018d3f9ee5dea17aa7849fcc9e71a/lib/backoffice/resolvers/ecto.ex#L206) which would prevent our mock repo from being used.

Reply:
Actually seems like this is not true. We need the `one` function but it can return 0.